File: frontend/src/types/Comment.ts
import type { User } from './User.js';
import type { Post } from './Post.js';

export interface Comment {
    id: number;
    content: string;
    active: boolean;
    created_at: string;
    updated_at: string;

    author: Pick<User, 'id' | 'username' | 'fullName'>;
    post: Pick<Post, 'id' | 'title'>;
}

File: backend/src/main/java/com/ufc/blog/entity/Comment.java
package com.ufc.blog.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "comments")
public class Comment extends AuditableEntity {

    @NotBlank
    @Size(max = 1000)
    @Column(name = "content", length = 1000, nullable = false)
    private String content;

    @Column(name = "active", nullable = false)
    private Boolean active = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Post post;

    public Comment(String content, User author, Post post) {
        this();
        this.content = content;
        this.author = author;
        this.post = post;
    }

}
